server_test.go

```diff
@@ -0,0 +1,179 @@
+package timetracker_test
+
+import (
+	"bytes"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+	"timetracker"
+)
+
+func TestGetEnvironmentVariables(t *testing.T) {
+	t.Parallel()
+
+	type testCase struct {
+		fn func(string) (string, error)
+		a  string
+	}
+	tcs := []testCase{
+		{fn: timetracker.GetEnvironmentVariable, a: "TIMETRACKER_DB_HOST"},
+		{fn: timetracker.GetEnvironmentVariable, a: "TIMETRACKER_DB_PORT"},
+		{fn: timetracker.GetEnvironmentVariable, a: "TIMETRACKER_DB_USER"},
+		{fn: timetracker.GetEnvironmentVariable, a: "TIMETRACKER_DB_NAME"},
+	}
+	for _, tc := range tcs {
+		_, err := tc.fn(tc.a)
+		if err != nil {
+			t.Fatalf("error with environment variable: %s %s", tc.a, err)
+		}
+	}
+
+}
+
+func TestServerRoutesStatusOK(t *testing.T) {
```

Comment:
Again, this doesn't really seem to test anything other than that the specified endpoints return `StatusOK`. If that's literally the _only_ behaviour they're supposed to have, fine—but then they wouldn't be much use!

On the other hand, if an endpoint is supposed to _do_ something (list your tasks, for example), then that's what we should be trying to test, isn't it? A table test definitely wouldn't be the right choice in that case, because we'd be looking for different behaviour from each endpoint. I suggest refactoring these into separate tests,  each designed to exercise a specific behaviour (for example, `TestCreateEndpointCreatesTask`).